pkg/manifestreader/common.go

```diff
@@ -102,26 +95,56 @@ func SetNamespaces(factory util.Factory, infos []*resource.Info,
 		case "Cluster":
 			continue
 		case "Namespaced":
-			inf.Namespace = defaultNamespace
 			accessor.SetNamespace(defaultNamespace)
 		}
 	}
 
 	return nil
 }
 
-// FilterLocalConfig returns a new slice of infos where all resources
+// FilterLocalConfig returns a new slice of Unstructured where all resources
 // with the LocalConfig annotation is filtered out.
-func FilterLocalConfig(infos []*resource.Info) []*resource.Info {
-	var filterInfos []*resource.Info
-	for _, inf := range infos {
-		acc, _ := meta.Accessor(inf.Object)
+func FilterLocalConfig(objs []*unstructured.Unstructured) []*unstructured.Unstructured {
+	var filteredObjs []*unstructured.Unstructured
+	for _, obj := range objs {
+		acc, _ := meta.Accessor(obj)
 		// Ignoring the value of the LocalConfigAnnotation here. This is to be
 		// consistent with the behavior in the kyaml library:
 		// https://github.com/kubernetes-sigs/kustomize/blob/30b58e90a39485bc5724b2278651c5d26b815cb2/kyaml/kio/filters/local.go#L29
 		if _, found := acc.GetAnnotations()[filters.LocalConfigAnnotation]; !found {
-			filterInfos = append(filterInfos, inf)
+			filteredObjs = append(filteredObjs, obj)
 		}
 	}
-	return filterInfos
+	return filteredObjs
+}
+
+// removeAnnotations removes any Path or Index annotations from the
+// resource.
+func removeAnnotations(n *yaml.RNode, annotations ...kioutil.AnnotationKey) error {
+	for _, a := range annotations {
+		err := n.PipeE(yaml.ClearAnnotation(a))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// kyamlNodeToUnstructured take a resource represented as a kyaml RNode and
+// turns it into an Unstructured object.
+func kyamlNodeToUnstructured(n *yaml.RNode) (*unstructured.Unstructured, error) {
```

Comment:
This is new and looks like it's the foundation of all our `ManifestReaders`. Should we add a unit test?

Reply:
So this is used by the tests in both the `PathManifestReader` and the `StreamManifestReader`, so I don't think we need to test this separately.